designer/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentService.scala

```diff
@@ -89,7 +110,8 @@ class DefaultComponentService(
     processService
       .getLatestRawProcessesWithDetails[ScenarioComponentsUsages](ScenarioQuery(isArchived = Some(false)))
       .map { processDetailsList =>
-        val componentsUsage = ComponentsUsageHelper.computeComponentsUsage(componentIdProvider, processDetailsList)
+        val componentsUsage =
+          ComponentsUsageHelper.computeComponentsUsage(processDetailsList, processingTypeAndComponentInfoToComponentId)
```

Comment:
Let's pass a `Map[ProcessingType, ModelDefinition]` into `ComponentsUsageHelper` and use `modelDefinitionByProcessingType(processingType).getComponent(info)` there. Thanks to that this technical trick around component usage cash won't infect other places of the code.